### PR TITLE
[fix](merge-cloud) fix load return invalid tablet state when MoW table under alter

### PR DESCRIPTION
**Pull request by liaoxin01**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by liaoxin01** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

File: be/src/cloud/cloud_rowset_builder.cpp [hunk 1]
```diff
@@ -37,7 +37,11 @@ Status CloudRowsetBuilder::init() {
 
     std::shared_ptr<MowContext> mow_context;
     if (_tablet->enable_unique_key_merge_on_write()) {
-        RETURN_IF_ERROR(std::dynamic_pointer_cast<CloudTablet>(_tablet)->sync_rowsets());
+        auto st = std::dynamic_pointer_cast<CloudTablet>(_tablet)->sync_rowsets();
```
Comment (gavinchou) [inline be/src/cloud/cloud_rowset_builder.cpp:40]: Can we use static cast here?

Comment (liaoxin01) [reply be/src/cloud/cloud_rowset_builder.cpp]: Yes

**Comment by liaoxin01** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by liaoxin01** [comment]
run buildall

**Comment by liaoxin01** [comment]
run buildall

**Pull request merged**